lib/dao.js

```diff
@@ -926,13 +926,13 @@ DataAccessObject.remove = DataAccessObject.deleteAll = DataAccessObject.destroyA
 DataAccessObject.removeById = DataAccessObject.destroyById = DataAccessObject.deleteById = function deleteById(id, cb) {
   if (stillConnecting(this.getDataSource(), this, arguments)) return;
   var Model = this;
-  
-  this.remove(byIdQuery(this, id).where, function(err) {
+
+  this.getDataSource().connector.destroy(this.modelName, id, function (err) {
     if ('function' === typeof cb) {
       cb(err);
     }
     if(!err) Model.emit('deleted', id);
-  });
+  }.bind(this));
```

Comment:
Maybe, we should fix loopback-connector-rest to accept the where on deleteAll instead of merging this change?

But how could rest connector do a "where" ?